react/features/app/actions.any.ts

```diff
@@ -83,4 +92,51 @@ export function reloadWithStoredParams() {
     };
 }
 
+/**
+ * Checks whether tokenAuthUrl is set, we have a jwt token that will expire soon
+ * and redirect to the auth url to obtain new token if this is the case.
+ *
+ * @param {Dispatch} dispatch - The Redux dispatch function.
+ * @param {Function} getState - The Redux state.
+ * @param {Function} failureCallback - The callback on failure to obtain auth url.
+ * @returns {boolean} Whether we will redirect or not.
+ */
+export function maybeRedirectToTokenAuthUrl(
+        dispatch: IStore['dispatch'], getState: IStore['getState'], failureCallback: Function) {
+    const state = getState();
+    const config = state['features/base/config'];
+    const { locationURL = { href: '' } } = state['features/base/connection'];
+
+    if (isTokenAuthEnabled(config)) {
+        // if tokenAuthUrl check jwt if is about to expire go through the url to get new token
+        const jwt = state['features/base/jwt'].jwt;
+        const expirationDate = getJwtExpirationDate(jwt);
+
+        // if there is jwt and its expiration time is less than 3 minutes away
+        // let's obtain new token
+        if (expirationDate && expirationDate.getTime() - Date.now() < 3 * 60 * 1000) {
+            const room = state['features/base/conference'].room;
+            const { tenant } = parseURIString(locationURL.href) || {};
+
+            getTokenAuthUrl(config, room, tenant, true)
+                .then((tokenAuthServiceUrl: string | undefined) => {
+                    if (!tokenAuthServiceUrl) {
```

Comment:
How could this happen, if we checked token auth was enabled?

Reply:
Hum ... yeah ... I copied it from another place ...

Reply:
This can happen if room parameter is missing.